sdp/processors/datasets/masc/aggregate_segments.py

```diff
@@ -0,0 +1,123 @@
+        self.crop_audio_segments = crop_audio_segments
+        self.output_audio_dir = output_audio_dir
+        self.input_segments_key = input_segments_key
+        self.verbose = verbose
+        self.output_text_key = output_text_key
+        self.output_duration_key = output_duration_key
+
+    def prepare(self):
+        if self.crop_audio_segments and self.output_audio_dir:
+            os.makedirs(os.path.join(self.output_audio_dir), exist_ok=True)
+
+    def process_dataset_entry(self, data_entry: dict):
+        if self.input_segments_key not in data_entry:
+            if self.verbose:
+                logging.info(f"No segements in a sample {data_entry[self.input_audio_filepath_key]}.")
+            return []
+
+        segments = data_entry[self.input_segments_key]
+        if len(segments) == 0:
+            return []
+        
+        audio_basename = os.path.basename(data_entry[self.input_audio_filepath_key]).split(".")[0]
+        agg_segments = []
+        aggregated_segment = {**segments[0]}
+        for segment in segments[1:]:
+            # checking if after adding segement it's duration will exceed `max_duration`
+            start_time = min(segment["start_time"], aggregated_segment["start_time"])
+            end_time = max(segment["end_time"], aggregated_segment["end_time"])
+            if end_time - start_time >= self.max_duration:
+                agg_segments.append(aggregated_segment)
+                aggregated_segment = {**segment}
+            else:
+                # updating aggregated segment text with correct order of segments.
+                if aggregated_segment["start_time"] < segment["start_time"]:
+                    aggregated_segment["text"] += f" {segment['text']}".strip()
+                else:
+                    aggregated_segment["text"] = f"{segment['text']} {aggregated_segment['text']}"
+                    
+                aggregated_segment["start"] = start_time    # updating aggregated segment start time
+                aggregated_segment["end_time"] = end_time   # updating aggregated segment end time
+        else:
+            # adding the last aggregated segment 
+            if aggregated_segment not in agg_segments:
+                agg_segments.append(aggregated_segment)
+            
+        audio = AudioSegment.from_wav(data_entry[self.input_audio_filepath_key])
+        for aggregated_segment in agg_segments:
+            aggregated_segment.update(data_entry)
+            
+            start_time = aggregated_segment.pop("start_time")
+            end_time = aggregated_segment.pop("end_time")
+            
+            aggregated_segment[self.output_duration_key] = end_time - start_time
+            aggregated_segment[self.output_splitted_audio_filepath_key] = os.path.join(self.output_audio_dir, f"{audio_basename}_{start_time}_{end_time}.wav")
+            
+            if self.crop_audio_segments:
+                try:
+                    get_audio_segment(
```

Comment:
rename from "get_audio_segment" to something like "save_audio_segment"?
and maybe rename the flag form "self.crop_audio_segments" to something like "self.save_final_audio_segments"?